File: python3/bron_kerbosch2_gpx.py
# coding: utf-8

from bron_kerbosch_pivot import pick_max_degree, pick_max_degree_localX, visit
from graph import UndirectedGraph
from reporter import Reporter


def bron_kerbosch2_gpx(graph: UndirectedGraph, reporter: Reporter):
    '''Bron-Kerbosch algorithm with pivot of highest degree within remaining candidates
       chosen from both candidates and excluded vertices (IK_GPX)'''
    if candidates := graph.connected_vertices():
        visit(graph=graph,
              reporter=reporter,
              initial_pivot_choice=pick_max_degree,
              further_pivot_choice=pick_max_degree_localX,
              candidates=candidates,
              excluded=set(),
              clique=[])

File: go/lib/bron_kerbosch2_gpx.go
package BronKerbosch

func bronKerbosch2gpx(graph *UndirectedGraph, reporter Reporter) {
	// Bron-Kerbosch algorithm with pivot of highest degree within remaining candidates
	// chosen from both candidates and excluded vertices (IK_GPX).
	candidates := graph.connectedVertices()
	if candidates.IsEmpty() {
		return
	}
	excluded := make(VertexSet, len(candidates))
	visit(graph, reporter, MaxDegree, MaxDegreeLocalX, candidates, excluded, nil)
}
